### ISSUE TITLE
Loading Checkpoint Fails

### ISSUE THREAD

**Issue by Alphastrick** [nicolas-chaulet/torch-points3d#676]
Hi there,

I'm working with VoteNet and tried to continue the aborted training.

```
python train.py task=object_detection model_name=VoteNetPaper models=object_detection/votenet2 data=object_detection/c3docbbds training.checkpoint_dir="/torch-points-3d/outputs/<checkpoint>/"
```

Unfortunately, this error occurred:
 
```
Traceback (most recent call last):
  File "train.py", line 14, in main
    trainer.train()
  File "/torch-points-3d/torch_points3d/trainer.py", line 144, in train
    self._train_epoch(epoch)
  File "/torch-points-3d/torch_points3d/trainer.py", line 198, in _train_epoch
    self._model.optimize_parameters(epoch, self._dataset.batch_size)
  File "/torch-points-3d/torch_points3d/models/base_model.py", line 246, in optimize_parameters
    self._grad_scale.step(self._optimizer)  # update parameters
AttributeError: 'NoneType' object has no attribute 'step'
```
Does anyone has a clue how to resolve this error?

**Comment by chrockey** [comment]
Hi I have the same problem. Any help will be grateful!

**Comment by ZheningHuang** [comment]
save here. Looking for an update

**Comment by LucaRom** [comment]
Hi, I faced the same issue but was able to fix it, so maybe the information can help to find the source of the error for others.

The error only occurred while using a docker image on remote server and only after the first run (first run would go through). Basically, the mounting of folders where wrong and it wouldn't create new config files for each run. So I think it would try to push hydra's config file with the new overrides but would fail and it somehow result in a **nonetype** being returned. The issue also happened if I would not mount the folders with the proper permissions; creating the output files in read-only mode. 

So basically, needed to create mounted folders properly and make sure I would create them in a place where I could have all permissions on the remote server. So if you are sure your mounts and permissions are good, my assumption would be that the problem is in the way the config files containing the parameters/overrides are handle at the beginning of the run.

**Comment by pfmephisto** [comment]
I'm also experiencing the same issue, although as far as I can tell all my directory permissions are fine.

What I can't understand from the code is where `_grad_scale` should be set when loading from a checkpoint.
So far the only place I have found where `_grad_scale` is being set is inside `instantiate_optimizers` which is only being called if no checkpoints are present.
https://github.com/torch-points3d/torch-points3d/blob/66e8bf22b2d98adca804c753ac3f0013ff4ec731/torch_points3d/trainer.py#L89-L105

Otherwise the `create_model` which is called on the checkpoint in turn calls `_initialize_model`, but nowhere down that line `_grad_scale` has ever been set and so it stays initialized as `None`
https://github.com/torch-points3d/torch-points3d/blob/66e8bf22b2d98adca804c753ac3f0013ff4ec731/torch_points3d/metrics/model_checkpoint.py#L170-L181

https://github.com/torch-points3d/torch-points3d/blob/66e8bf22b2d98adca804c753ac3f0013ff4ec731/torch_points3d/metrics/model_checkpoint.py#L225-L229

https://github.com/torch-points3d/torch-points3d/blob/66e8bf22b2d98adca804c753ac3f0013ff4ec731/torch_points3d/metrics/model_checkpoint.py#L93-L121

I would assume that somewhere in `load_optim_sched` something akin to the following should be set.
https://github.com/torch-points3d/torch-points3d/blob/66e8bf22b2d98adca804c753ac3f0013ff4ec731/torch_points3d/models/base_model.py#L327
